README.md
# 本项目为毕节市公积金后端项目，项目的代码架构视图为
## 1.　housingfund　作为项目的根目录，作为父项目，囊括所有的子项目
## 2.　housingfund-collection　为公积金归集项目业务模块
## 3.　housingfund-common 为整个公积金项目的通用模块，包含RPC实现，数据库工具实现等
## 4.　housingfund-bank　为公积金的结算平台接口模块
## 5.　housingfund-bank-server　为公积金的结算平台到账通知模块
## 6.　housingfund-loan　为公积金项目的贷款业务模块
## 7.  housingfund-server　为公积金项目的API服务模块
## 8.　housingfund-withdrawls　为公积金项目的提取业务模块
## 9.  housingfund-account 公积金项目权限管理
## 10.  housingfund-logger 　公积金项目日志管理
## 11. housingfund-database 公积金数据库层
## 12. housingfund-others 其他模块
## 13. housingfund-ca-service 为深圳CA中心提供的webservice接口模块
## 14  housingfund-statemachine 为状态机实现
## 15  housingfund-task 定时任务模块实现


#　将项目根的setting.xml文件拷贝到  ~/.m2/    目录下　　　

git修改.gitignore文件，不生效解决步骤，依次在工程目录下执行下面命令：

```
git rm -r -f --cached .

git add .

git commit -m 'update .gitignore'
```

ChangeLog.md
# v1.2.28
>2018-03-13
* 优化 基数调整导出模板：过滤封存状态的用户
* 优化 缴存记录页面字段“未分摊合计”、“单位账户余额合计”结果随搜索条件变化

# v1.2.27
>2018-03-09
* 修复预封存的历史缴存基数不能修改、显示有误的问题
* 贷款账户查询列表新增“贷款发放额”字段、新增贷款余额总合计、本页合计、贷款发放总合计、本页合计
* 对应调基导入、导出excel调整（注意替换personRadix.xlsx模板）
* 优化、校正 财务的全市归集贷款统计表、缴存提取分类情况表
* 合同变更字段“个人账户状态”实时获取状态

# v1.2.26
>2018-03-05
* 优化 贷款合同回执单中房屋坐落的显示（注意替换LoanContract模板）
* 优化 合同变更回执单的内容（注意替换ContractAlterTemp模板）
* 新增 销户提取提示功能：检查封存状态下缴至年月问题
* 修复预封存的历史缴存基数不能修改、显示有误的问题（只记录，暂不更新）

# v1.2.25
>2018-03-02
* 新增 异地贷款职工住房公积金缴存证明PDF（更新sql，新增DiffTerritoryLoadProve模板，collection修改dubbo文件73行）
* 修复 合同变更必要数据为空的问题
* 修复 40103到4010301
* 新增 委托扣款协议打印（更新模板LoanEntrustDeduct.pdf，sql权限）

# v1.2.24
>2018-02-28
* 新增 个人账户信息详情中贷款信息的姓名字段
* 新增 异地贷款职工住房公积金缴存证明PDF（更新sql，新增DiffTerritoryLoadProve模板，collection修改dubbo文件73行）（已取消）
* 合户优化
* 扣款失败列表加身份证
* 转移bug
* 提取修复
* 修复审核流程
* 贷款申请加网点搜索
* 银行对账增加了实时余额

# v1.2.23
>2018-02-06
* 修改 暂收汇缴为未分摊汇补缴 (更新sql)
* 修改 未分摊记录列表金额搜索忽略正负号
* 修复 日常财务处理中备注查询问题
* 添加 还款序号解决还款重复匹配问题
* 修复 从单位转出后在转入原单位，清册时没这个人的数据
* 新增 全市归集统计表未分摊余额统计（更新dubbo.xml，sql）
* 修改 归集封存原因
* 修复 开户产生补缴，再进行封存产生错缴，此时未生成错缴记录
* 修复 个人错缴后，缴至年月更新为正常汇缴的最新年月
* 修改 在贷款审核界面能看到上一级审批，弄在界面显示，不用点进流水号就能看到
* 修改 提取还贷，共同借款人公积金账户非正常时，报必要数据为空的问题
* 修改 失败原因叠加问题
* 新增 贷款的变更信息列表，添加网点查询
* 修改 贷款初始状态问题
* 修改 同一个单位下，封存个人后合户到另一个人，清册时个人账户状态错误	

# v1.2.22
>2018-02-02（已部署
* 新增 基数列表个人月缴存额，单位月缴存额，个人月缴存额合计，单位月缴存额合计，月缴存额合计字段
* 修改 缴存业务回执单盖章位置（头部）
* 修改 个人缴存明细根据时间范围打印
* 修改 提取回执单盖章数量（替换WithdrawlReceiptTemp模板）
* 优化 个人导入时，证件号码错误提示修改为身份证号码不正确
* 优化 解决重复发往结算平台问题
* 优化 同一个单位下，封存个人后合户到另一个人，清册时个人账户状态错误	
* 新增 未分摊记录这里，增加一列“未分摊余额”，未分摊记录”来源增加【暂收汇缴】
* 优化 在缴存清册体现单位部分和个人部分
* 新增 合同变更审核列表-借款人姓名字段
* 新增 个人转移时显示原单位的基数

# v1.2.21
>2018-01-31（已部署
* 凭证查询页面新增摘要查询
* 修复SettlementHandler中由于结算平台异常而导致的空指针异常
* 单位缴存列表增加按缴至年月查询
* 选择未分摊时倒序排序
* 楼盘信息中预售证为一栋一证，把预售许可证提到楼栋号之前填写
* 新增 还款计划pdf接口(新增RepaymentPlanTemp-0、RepaymentPlanTemp-1模板，新增权限sql)
* 新增 导出还款记录excel接口（other模块中dubbo文件新增引用loanAccountService，新增PaymentHistoryTemp模板，新增权限sql）
* 优化 导出还款记录模板，修改代码（替换PaymentHistoryTemp模板）

# v1.2.20
>2018-01-30（已部署
* 修正全市统计表时间范围不准确的BUG
* 提取与转移在途验证
* 扣款失败记录列表，新增网点查询功能
* 签订合同列表添加操作员信息

# v1.2.19
>2018-01-29（已部署
* 添加还款序号解决还款重复匹配问题
* 修复 日常财务处理列表查询报错问题
* 其他提取时可选择销户或部分
* 提取列表新增发生额和发生利息额
* 还款申请，待入账业务添加删除功能

# v1.2.18
>2018-01-25（已部署
* 需求变更 更改到账匹配优先级为：备注>摘要>对手户名
* 修复 贷款申请审批表空指针的问题
* 职工列表导出速度优化
* 新增 提取业务列表按“总额”查询
* 修复 合同变更中心人员看不到业务的BUG
* 签订合同列表新增状态和金额搜索，扣款失败记录新增业务类型搜索

# v1.2.17
>2018-01-24（已部署
* 新增 归集的职工列表
* 新增 导出职工列表excel (注：今天不更新，需优化)
* 修改 导出清册数据排序
* 修改 公积金账号不存在时，审批表PDF也要显示共同借款人姓名、证件号码
* 修复离异后提取时，过滤婚姻状况
* 修复销户提取时，对汇缴业务进行在途验证，过滤掉已封存的职工
* 新增 签订合同、合同变更时还款账户户名字段
* 优化 导出职工列表效率
* 修改 导出职工列表数据排序
* 修改 调基调比列表提交不生效的问题

# v1.2.16
>2018-01-23（已部署
* 修复提取办结操作报必要数据为空的BUG
* 个人变更审核走受理网点流程
* 个人变更业务列表添加按网点搜索
* 提取业务修正在途验证

# v1.2.15
>2018-01-19（已部署
* 贷款撤回功能调整为：贷款放款之前都可以作废，删除
* 暂收记录列表新增 暂收记账凭证
* 新增 导出清册确认单excel
* 日常财务处理列表，增加通过备注查询功能
* 更新审批表模板
* 增加银行日记账，银行流水重复验证
* 贷款 还款记录显示余额
* 修复 未分摊转移摘要bug
* 修复 调基、调比：取消审核流程，直接办结未生效
## 提取修改
* 其他或死亡提取时，可变更收款银行
* 提取还贷时，借款人与共同借款人可同时提取
* 提取还贷审核通过时，还贷记录产生，提取状态未变化
* 提取业务在办理网点审核，不再发往开户网点审核
* 提取时验证是否有未完成的汇补缴业务
* 修改0余额0利息销户提取不走结算平台
* 提取列表网点搜索修复


# v1.2.14
>2018-01-18（已部署
* 借款人提取还款，取消共同借款人验证
* 修复 贷款结清时计算误差问题



# v1.2.13
>2018-01-17（已部署
* 账户变动通知单摘要搜索里增加备注搜索
* 新增贷款账户状态 已作废
* 修复 打印个人缴存明细pdf盖章位置
* 优化 导入基数调整失败时提示信息
* 新增 批量打印财务凭证pdf
* 去除贷款还款申请检测公积金账号的限制

# v1.2.12
>2018-01-16（已部署
* 优化借款合同模板，修改页码，修改地28页内容
* 删除审批表最后一页的承诺内容
* 归集定时任务
* 提取还贷关联公积金账户销户问题（修复 9915082700002代远志 提取还贷问题）
* 新增还款申请只能当天审核的限制
* 修复 打印个人缴存明细pdf盖章位置
* 优化 导入基数调整失败时提示信息
* 新增 批量打印财务凭证pdf
* 新增 封存时提取的在途验证
* 合同pdf银行名称填充

# v1.2.11
>2018-01-15（已部署
* 到账通知单列表查询时间改为created_at
* 更新贷款合同模板，更改页码，删除表格多余的两行
* 提前还款期数问题

# v1.2.10
>2018-01-12（已部署
* 修复 归集-个人开户导入功能
* 修复 归集-清册比例显示问题
* 修复 归集-汇缴待入账是清册问题
* 修复 归集-个人导入联名卡问题
* 新增 贷款-扣款失败保留扣款失败原因
* 修复 贷款-打印结清报错
* 修复 贷款-接口合同保证人信息错误
* 修复 归集-基数导入未去空格
* 修复 归集-调基办结效率优化
* 修复 归集-调比办结效率优化
* 修复 贷款-大额让你行关闭导致无法入账问题
* 修复 贷款-修复单笔收款是type被标记为3的bug
* 修复 归集-基数导入：单位账号、单位名称有空格时，提示【个人账号不属于该单位】
* 财务日常财务处理中未分摊转移取消销户单位限制

# v1.2.9
>2018-01-10（已部署
* 修复 公积金使用情况表-》优化所有查询
* 修复 全市归集贷款统计表-》归还贷款-》按业务进行统计分开
* 修复 住房公积金缴存提取分类情况表
* 新增 基数导出增加身份证号
* 新增 打印结清证明

# v1.2.8
>2018-01-09（已部署
* 优化了还款大部分sql查询语句，添加了业务网点字段
* 银行存款情况表-》统计住房公积金存款合计和增值收益存款合计
* 公积金使用情况表-》优化所有查询
* 全市归集贷款统计表-》归还贷款-》按业务进行统计分开
* 增加 住房公积金缴存提取分类情况表
* 修复 提取还贷产生逾期问题

# v1.2.7
>2018-01-08（已部署
* 提取还贷验证银行名称（取消验证
* 调整基数采用翻页的方式及优化效率问题 
* 共同借款人提取还贷报错“证件号码不存在或账户已结清，查询账户状态后再办理
* 提取还贷提示信息优化


# v1.2.6
>2018-01-08（已部署
* 修复财务日常财务处理通用bug
* 修复基数导入前后不一致问题
* 修复提取还贷问题

# v1.2.5
>2018-01-04（已部署
* 紧急修复贷款审核问题


# v1.2.4
>2018-01-03（已部署
* 列表查询新增字段
* 修改合同变更pdf数据
* 暂收列表查询问题
* 修正导入个人账户功能手机号码
* 结账问题
* 贷款优化


# v1.2.3
>2018-01-02（已部署
* 日常财务处理列表查询日期bug修复
* 日常财务处理新增合计、备注字段
* 日常财务处理-》通用模块增加绑定到账通知单功能
* 贷款优化

# v1.2.2
>2017-12-28（已部署
* 解决信用社转预约执行导致提取入账失败问题
* 未分摊记录新增错缴类型
* 银行机构管理问题
* 借贷不平衡限制
* 解决中行主机流水号问题
* 新增到账通知去重功能
* 贷款优化

# v1.2.1
>2017-12-22（已部署
* 双贯标验收版本

# v1.1.1
>2017-12-12（已部署
* 提取手动处理做账问题
* 对账功能优化
* 单位汇缴根据名称搜索报错问题


# v1.1.0
>2017-12-11（已部署
* 贷款功能上线


# v1.0.41
>2017-12-02（已部署
* 跨行、同行转账问题修复，记账凭证和到账通知单改为1对多
* 11月29号客户提出的问题滴41条，42条，43条，44条，45条bug修复
* 新增转账汇缴
* 缴存基数调整表增加身份证号


housingfund-common/src/main/java/com/handge/housingfund/common/Constant.java
package com.handge.housingfund.common;

/**
 * Created by xuefei_wang on 17-8-31.
 */
public final class Constant {

    public static final String JWT_CONF = "jwt";

    public static final String ACCOUNT_CONF = "account";

    public static final String COMMON_CONF = "common" ;

    public static final String SERVER_CONF = "server" ;

    public static final String FINANCE_CONF = "finance";

    public static final String LOAN_CONF = "loan";

    public static final String BANK_CONF = "bank";

    public static final String TASK_CONF = "task";


    public final static class JWT {
     public final static String  secret = "secret";
     public final static String  secret_defualt = "jkhKLJSDFfsmbfsJKHSDFdf897891j3hjgvgf13g2jhg312313";
     public final static String  issuer = "issuer";
     public final static String  issuer_defualt = "handge.com";
     public final static String  expires = "expires";
     public final static long    expires_defualt = 240000l;
    }

    public final static class Accout {


    }

    public final static class Common {


    }

    public final static class Server {

        public static  final  String  TOKEN_KEY = "TOKEN_CONTEXT";

        public static final  String TOKEN_HEAD = "H-TOKEN";

        public static final  String ADMIN_TOKEN = "admin_token";

        public static final String no_token_urls = "notoken_urls";

        public static final String token_expires = "token_expires";

        public static final int token_expires_defualt = 15;


    }

    public final static class Finance {


    }

    public final static class Loan {


    }

    public final static class Bank {

    }

}
